packages/server/src/insights/insights.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  QuestionType,
+  Role,
+  PossibleOutputTypes,
+  SimpleDisplayOutputType,
+  SimpleChartOutputType,
+  InsightDisplay,
+} from '@koh/common';
+import { UserCourseModel } from 'profile/user-course.entity';
+import { SelectQueryBuilder } from 'typeorm';
+import { QuestionModel } from 'question/question.entity';
+
+export interface InsightInterface<Model> {
+  name: string;
+  displayName: string;
+  description: string;
+  roles: Role[];
+  component: InsightDisplay;
+  model: new () => Model; // One of the modals have
+  possibleFilters: string[];
+  compute: (
+    queryBuilder: SelectQueryBuilder<Model>,
+    insightFilters: any,
+  ) => Promise<PossibleOutputTypes>;
+  output?: Promise<PossibleOutputTypes>;
+}
+
+function addFilters(
+  queryBuilder,
+  modelName,
+  filters,
+  possibleFilters,
+): SelectQueryBuilder<QuestionModel> {
+  for (const filter of filters) {
+    if (!possibleFilters.includes(filter.type)) {
```

Comment:
what's preventing

```
for (const filter of filters) {
   if (possibleFilters.includes(filter.type)) {
       FILTER_MAP[modelName][filter.type](queryBuilder, filter);
   }
}
```

I feel like that'd be cleaner!

Reply:
You're totally right, just got that changed.